File: jscomp/test/debugger_test.ml
let f x y =
  Js.log (x, y) ;
  x + y

let g () =
  ignore @@ f 1 2 ;
  [%bs.debugger] ;
  ignore @@ f 1 2 ;
  [%bs.debugger] ;
  3

let exterme_g () =
  ignore @@ f 1 2 ;
  let v = [%bs.debugger] in
  Js.log v ;
  ignore @@ f 1 2 ;
  [%bs.debugger] ;
  3

File: jscomp/test/debugger_test.js
'use strict';


function f(x, y) {
  console.log(/* tuple */[
        x,
        y
      ]);
  return x + y | 0;
}

function g() {
  f(1, 2);
  debugger;
  f(1, 2);
  debugger;
  return 3;
}

function exterme_g() {
  f(1, 2);
  debugger;
  var v = /* () */0;
  console.log(v);
  f(1, 2);
  debugger;
  return 3;
}

exports.f = f;
exports.g = g;
exports.exterme_g = exterme_g;
/* No side effect */
